core/types/l2trace.go

```diff
@@ -21,9 +21,11 @@ var (
 	}
 )
 
-// BlockResult contains block execution traces and results required for rollers.
-type BlockResult struct {
-	BlockTrace       *BlockTrace        `json:"blockTrace"`
+// BlockTrace contains block execution traces and results required for rollers.
+type BlockTrace struct {
+	Coinbase         *AccountWrapper    `json:"coinbase"`
+	Header           *Header            `json:"header"`
```

Comment:
we need to add a list of block hashes for `BLOCKHASH` opcode to use, starting from the current block.

Reply:
The `BLOCKHASH` opcode is supported by this [func](https://github.com/scroll-tech/go-ethereum/blob/0bdd2e0575a2b5741b757a9966120ef83ecddbad/core/evm.go#L57).